src/polytopes/segment.jl

```diff
@@ -36,15 +36,18 @@ end
 
 function Base.in(p::Point{Dim,T}, s::Segment{Dim,T}) where {Dim,T}
   a, b = s.vertices
-  if Dim == 2
-    _0 = zero(T)
-  elseif Dim == 3
-    _0 = Vec{3,T}(0, 0, 0)
-  else
-    throw(ErrorException("not implemented"))
+  ab, ap = b - a, p - a
+  # points a, b, p are collinear if and only if the cross-products (cp) for ab and ap
+  # with respect to all possible pairs of coordinates are zero
+  iscollinear = true
+  for i in 1:Dim, j in (i+1):Dim
+    cp = Vec{2,T}(ab[[i, j]]) × Vec{2,T}(ap[[i, j]])
```

Comment:
You rejected the suggestion here, but I think we need to avoid this allocation with [i,j]. Can you please explain why the suggestion was not applied?